lib/msf/core/post/file.rb

```diff
@@ -638,6 +642,153 @@ def copy_file(src_file, dst_file)
+      final_permissions_array << icacls_groups_permissions_array[i][0].gsub(/[()]/, '').split(',')
+    end
+    final_permissions_array
+  end
+
+  #
+  # Find what groups the current user is a member of.
+  #
+  # @return [Array] Array of groups the current user is a member of.
+  def win_find_current_user_groups
+    cuser_groups = cmd_exec('whoami /groups').to_s
+    cuser_groups_array = cuser_groups.scan(/^(?:\w+ ?[\w\-]*+ ?[\w\-]*+\\?[\w\-]*+ ?[\w\-]*+ ?[\w\-]*+ ?)/)
+    cuser_groups_length = cuser_groups_array.length
+    for i in 0...cuser_groups_length
+      cuser_groups_array[i].strip! # Strip any leading or trailing characters off the groups captured by the regex.
+    end
+
+    # Remove erronious entries that don't belong here but are captured by the regex.
+    cuser_groups_array.delete('GROUP INFORMATION')
+    cuser_groups_array.delete('Group Name')
+    cuser_groups_array
+  end
+
+  #
+  # See if +path+ on the remote system is accessible in the manner specified by +perm+ by the current user.
+  #
+  # @param path [String] Remote path to check
+  # @param perms [Array] Array of permissions to check for
+  #
+  # @return [Boolean] true if +path+ is accessible to the current user via any of the manners listed in +perms+, otherwise false.
+  def check_win_path_permissions(path, perms)
+    # Next check if icacls exists as a command on the target system, and switch to calcs if not.
+    if command_exists?('icacls')
+      # Grab permissions of the path in question and extract relevant info using a regex.
+      # Also strip extra whitespace before comparison occurs.
+      icacls_output_raw = cmd_exec("icacls \"#{path}\"").to_s
+
+      # Perform the actual regex extraction to determine what groups are permitted to
+      # access the file and what permissions each of those groups has on the file.
+      # For reference this is what the output from running "icacls" on win32kfull.sys looks like:
+      #
+      # C:\Users>icacls C:\Windows\System32\win32kfull.sys
+      # C:\Windows\System32\win32kfull.sys NT SERVICE\TrustedInstaller:(F)
+      #                                    BUILTIN\Administrators:(RX)
+      #                                    NT AUTHORITY\SYSTEM:(RX)
+      #                                    BUILTIN\Users:(RX)
+      #                                    APPLICATION PACKAGE AUTHORITY\ALL APPLICATION PACKAGES:(RX)
+      #
+      # Successfully processed 1 files; Failed processing 0 files
+      #
+      # C:\Users>
+      icacls_groups_array = grab_icacls_groups(icacls_output_raw)
+      icacls_groups_permissions_array = grab_icacls_groups_permissions(icacls_output_raw)
+
+      # Now that we have both the groups and the associated permissions for each group, zip this info together
+      # into a multidimensional array so that each group is contained with its corresponding permissions.
+      permissions_file_array = icacls_groups_array.zip(icacls_groups_permissions_array)
```

Comment:
```suggestion
      file_icacls_groups = parse_icacls_output(icacls_output_raw)
```
Possible return value:
```
{
 'NT SERVICE\TrustedInstaller' => ['F'],
 'BUILTIN\Administrators' => ['RX'],
 'USER1' => [ 'D', 'WDAC' ]
}
```

Reply:
```suggestion
      file_icacls_groups = find_icacls_permissions(path)
```

Reply:
Suggested change implemented in https://github.com/rapid7/metasploit-framework/pull/15987/commits/21be7a0ee8a0de1cc3ab01a8d117819122a89120